core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -54,7 +54,31 @@ enum SchemaVersion {
   static final String TABLE_RECORD_TYPE = "TABLE";
   static final String VIEW_RECORD_TYPE = "VIEW";
 
-  private static final String V1_DO_COMMIT_SQL =
```

Comment:
Do we also need to handle the same for `V1_DO_COMMIT_CREATE_SQL` ?

Reply:
It's not necessary as it creates the record, so populating cleanly the `iceberg_type`. The problem is just with the update where `iceberg_type` is used in the `WHERE` clause.

Reply:
Yeah, True. I just saw the pattern and commented (could have checked more). It is not needed here as it is an insert statement and not the query.